app/services/katello/pulp/status.rb

```diff
@@ -0,0 +1,23 @@
+module Katello
+  module Pulp
+    class Status < ::ProxyStatus::Status
+      def initialize(proxy, opts = {})
+        @api ||= Katello::ProxyAPI::Pulp.new(:url => proxy.url)
+        super proxy, opts
+        add_cache_key cache_key
+      end
+
+      def pulp_status
+        Rails.cache.fetch(cache_key, :expires_in => cache_duration) do
+          fetch_proxy_data do
+            @api.status
+          end
+        end
+      end
+
+      def cache_key
+        "proxy_#{proxy.id}/pulp"
+      end
+    end
+  end
+end
```

Comment:
This class looks like it talks to the smart proxy directly? If so, I think this would be better in a `services/katello/smart_proxy` directory structure given this directory is for direct communication to Pulp.